src/js/internal/fs/glob.ts

```diff
@@ -0,0 +1,75 @@
+import type { GlobScanOptions } from "bun";
+const { validateObject, validateString, validateFunction } = require("internal/validators");
+
+const isWindows = process.platform === "win32";
+
+interface GlobOptions {
+  /** @default process.cwd() */
+  cwd?: string;
+  exclude?: (ent: string) => boolean;
+  /**
+   * Should glob return paths as {@link Dirent} objects. `false` for strings.
+   * @default false */
+  withFileTypes?: boolean;
+}
+
+interface ExtendedGlobOptions extends GlobScanOptions {
+  exclude(ent: string): boolean;
+}
+
+async function* glob(pattern: string | string[], options: GlobOptions): AsyncGenerator<string> {
+  pattern = validatePattern(pattern);
+  const globOptions = mapOptions(options);
+  let it = new Bun.Glob(pattern).scan(globOptions);
+  const exclude = globOptions.exclude;
+
+  for await (const ent of it) {
+    if (exclude(ent)) continue;
+    yield ent;
+  }
+}
+
+function* globSync(pattern: string | string[], options: GlobOptions): Generator<string> {
+  pattern = validatePattern(pattern);
+  const globOptions = mapOptions(options);
+  const g = new Bun.Glob(pattern);
+  const exclude = globOptions.exclude;
+  for (const ent of g.scanSync(globOptions)) {
+    if (exclude(ent)) continue;
```

Comment:
followup for the no-op `no` function. i feel like if an exclude is not provided, then the result of g.scanSync can be returned directly, instead of wrapping it in a for loop that yields all results. [`g.scanSync` and `scan` both are iterators](https://github.com/oven-sh/bun/blob/29839737dffa92b63dd96619bc9079fe14e9849b/src/js/builtins/Glob.ts#L6-L21)

```ts
const exclude = options.exclude; // not globOptions
const scan = g.scanSync(globOptions);
if (options.exclude) {
   // existing filter
}
return scan;
```

apologies for not spotting this in the original review.